mlir/lib/Dialect/MIOpen/utility/builderUtils.cpp

```diff
@@ -80,5 +81,71 @@ Value createZeroConstantOp(OpBuilder &b, Location loc, Type type) {
   }
 }
 
+//===----------------------------------------------------------------------===//
+// Utility function to emit type conversion ops.
+//===----------------------------------------------------------------------===//
+Value createTypeConversionOp(OpBuilder &b, Location loc, Value source,
+                             Type sourceType, Type destType) {
+  // Convert from sourceType to destType if necessary.
+  Value result = source;
+  Type sourceElemType = sourceType;
```

Comment:
`Type sourceType = source.getType();` above

Also, cleanup note since we're touching this code: all of this could be simplified with `mlir::getElementTypeOrSelf()`